src/adapter/src/coord/timestamp_selection.rs

```diff
@@ -358,7 +365,16 @@ pub trait TimestampProvider {
         if when.can_advance_to_upper()
             && (isolation_level == &IsolationLevel::Serializable || timeline.is_none())
         {
-            candidate.join_assign(&largest_not_in_advance_of_upper);
+            let mut timestamp = largest_not_in_advance_of_upper;
+            // In practice advancing the timestamp into the future is confusing for users. So if
+            // there is an `oracle_read_ts` then we treat is as "the current time" bound the upper
```

Comment:
Is this sentence missing a word? "bound" seems to need something before or after it.

Reply:
Ah, it's missing the word "and". It should be
```
            // In practice advancing the timestamp into the future is confusing for users. So if
            // there is an `oracle_read_ts` then we treat is as "the current time" and bound the
            // upper advancement to "the current time".
```

Just updated.